addons/survey/tests/test_survey.py

```diff
@@ -144,6 +144,66 @@ def test_partial_scores_simple_choice(self):
         self.assertEqual(user_input.scoring_percentage, 100)
         self.assertTrue(user_input.scoring_success)
 
+    def test_simple_choice_question_answer_result(self):
+        test_survey = self.env['survey.survey'].create({
+            'title': 'Test This Survey',
+            'scoring_type': 'scoring_with_answers',
+            'scoring_success_min': 80.0,
+        })
+        [a_01, a_02, a_03, a_04] = self.env['survey.question.answer'].create([{
+            'value': 'In Europe',
+            'answer_score': 0.0,
+            'is_correct': False
+        }, {
+            'value': 'In Asia',
+            'answer_score': 5.0,
+            'is_correct': True
+        }, {
+            'value': 'In South Asia',
+            'answer_score': 10.0,
+            'is_correct': True
+        }, {
+            'value': 'On Globe',
+            'answer_score': 5.0,
+            'is_correct': False
+        }])
+        q_01 = self.env['survey.question'].create({
+            'survey_id': test_survey.id,
+            'title': 'Where is india?',
+            'sequence': 1,
+            'question_type': 'simple_choice',
+            'suggested_answer_ids': [(6, 0, (a_01 | a_02 | a_03 | a_04).ids)]
+        })
+
+        user_input = self.env['survey.user_input'].create({'survey_id': test_survey.id})
+        user_input_line = self.env['survey.user_input.line'].create({
+            'user_input_id': user_input.id,
+            'question_id': q_01.id,
+            'answer_type': 'suggestion',
+            'suggested_answer_id': a_01.id
+        })
+
+        def assertOnlyAnswerInStatsIs(status, stats):
+            ''' Assert counts for 'Correct', 'Partially', 'Incorrect', 'Unanswered' are 0, and 1 for status one '''
+            self.assertTrue(all([
+                total['count'] == int(total['text'] == status) for total in stats['totals']
+            ]), f'The only answer should be {status} answer for this user input')
```

Comment:
So we are casting a boolean to int to get 0 or 1 ? A little magic.

Would be a bit easier to read with:

```py
for status, count in [(total['text'], total['count']) for total in stats['totals']]:
    self.assertEqual(count, 1 if status == expected_status else 0)
```